In [1]:
#import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#import my workout data csv called "Workout_Data"
df = pd.read_csv('workout_data.csv')

df.head()

Date Workout Name          Exercise Name  Set Order  Weight  \
0  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          1    90.0   
1  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          2   130.0   
2  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          3   130.0   
3  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          4   140.0   
4  2019-09-17 17:52:12      3. Legs              Leg Press          1   145.0   

  Weight Unit  Reps  RPE  Distance Distance Unit  Seconds Notes Workout Notes  \
0         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   
1         lbs   5.0  NaN       NaN           NaN        0   NaN           NaN   
2         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   
3         lbs   5.0  NaN       NaN           NaN        0   NaN           NaN   
4         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   

  Workout Duration  
0              31m  
1              31m  
2              31m  
3              31m  
4              31m

In this notebook I'm going to ask certain questions of the data set. 

First, how many unique days exist in the data set? Said another way, how much time does this dataset encompass? From the previous cell, the date field is a string that includes the time stamp as well. I need to separate those 2 pieces of info.

In [2]:
#This identifies the data type of the date column. 

print(df["Date"].dtype)

#creates a shortened version of the date string, only including the mmddyyy portion.
df['short_Date'] = df['Date'].str.slice(0,10)

df.head()


object


Date Workout Name          Exercise Name  Set Order  Weight  \
0  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          1    90.0   
1  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          2   130.0   
2  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          3   130.0   
3  2019-09-17 17:52:12      3. Legs  Squat (Smith Machine)          4   140.0   
4  2019-09-17 17:52:12      3. Legs              Leg Press          1   145.0   

  Weight Unit  Reps  RPE  Distance Distance Unit  Seconds Notes Workout Notes  \
0         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   
1         lbs   5.0  NaN       NaN           NaN        0   NaN           NaN   
2         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   
3         lbs   5.0  NaN       NaN           NaN        0   NaN           NaN   
4         lbs   6.0  NaN       NaN           NaN        0   NaN           NaN   

  Workout Duration  short_Date  
0              31m  2019-09-17  
1              31m  2019-09-17  
2              31m  2019-09-17  
3              31m  2019-09-17  
4              31m  2019-09-17

In [3]:
#unique dates is calculated by using nunique() which is a count of all the unique values in a data column. IN this case it's 
# date field.

unique_dates = df["short_Date"].nunique()

print(unique_dates)


218


This shows that I have 218 days of workout data in this dataset, which means that I was working out for 218 days. If I was working out everyother day, I could assume that I was working out around 4 days a week so I could divide this number by 4 and get the number of weeks (and then months) this data set includes.

In [4]:
numberofWeeks = unique_dates/4

print(numberofWeeks)

54.5


According to this number, this workout data includes over a full year of data. To confirm this, I can isolate the year in the date field and count how many unique values are there.

In [5]:
df["year"] = df['short_Date'].str.slice(0,4)


unique_Year = df["year"].nunique()
print(df["year"])
print(unique_Year)

0       2019
1       2019
2       2019
3       2019
4       2019
        ... 
4219    2021
4220    2021
4221    2021
4222    2021
4223    2021
Name: year, Length: 4224, dtype: object
3


This makes sense, starting in 2019 and going all the way through 2020 and eventually exporting the data sometime in 2021. 